### PR TITLE
AP-4066 add client means caption

### PR DESCRIPTION
**Pull request by RoseSAK**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4066)

**Add client means assessment caption to pages throughout the client means journey**
- For pages with forms, add the caption using the caption attribute of the form helper method
- For pages without forms, the caption is rendered via the page template tag, you can now pass a caption as an argument to this tag
- Form partials were removed for several pages in the means income journey - identify types of income/outgoings, cash incomes/outgoings and student finance. This is because they used to be used for both the citizen journey and provider journey but are no longer used for the citizen journey. Removing them simplifies the view and content yaml files
- Updated feature and unit tests to account for view changes and content removal/changes

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/providers/applicant_employed/index.html.erb [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
     <%= form.govuk_check_boxes_fieldset :applicant,
                                         legend: { text: page_title, size: "xl", tag: "h1" },
+                                        caption: { text: t("generic.client_means_caption"), size: "l" },
```
Comment (naseberry) [inline app/views/providers/applicant_employed/index.html.erb:17]: Discussion: the design pattern generally as both heading and caption at the same size, what are our reasons for differing from this pattern?

File: app/views/providers/open_banking_consents/show.html.erb [hunk 2]
```diff
@@ -7,19 +7,17 @@
 
     <%= page_template page_title: t(".heading"), form:, template: :basic do %>
 
-      <h1 class="govuk-heading-xl"><%= t(".heading") %></h1>
-
       <%= form.govuk_radio_buttons_fieldset(:provider_received_citizen_consent,
-                                            legend: { size: "m", tag: "h2", text: t(".online_banking_question") }) do %>
+                                            caption: { text: t("generic.client_means_caption"), size: "l" },
+                                            classes: ["govuk-!-margin-bottom-5"],
```
Comment (naseberry) [inline app/views/providers/open_banking_consents/show.html.erb:12]: does the option `classes` require the square brackets?

File: app/views/providers/open_banking_guidances/show.html.erb [hunk 3]
```diff
@@ -7,14 +7,15 @@
 
   <%= page_template page_title: t(".heading_h1"), form:, template: :basic do %>
 
+    <span class="govuk-caption-l"><%= t("generic.client_means_caption") %></span>
```
Comment (naseberry) [inline app/views/providers/open_banking_guidances/show.html.erb:10]: Discussion: the design pattern generally as both title and caption at the same size, what are our reasons for differing from this pattern?

File: app/views/providers/applicant_employed/index.html.erb (hunk 1 above)
Comment (RoseSAK) [reply app/views/providers/applicant_employed/index.html.erb:17]: Not sure, I was just following the sizing in the design/prototype

Comment (colinbruce) [reply app/views/providers/applicant_employed/index.html.erb:17]: We did the same on the proceeding loop caption-l H1-xl
Maybe we should raise this with design so we have a consistent change?

Comment (colinbruce) [reply app/views/providers/applicant_employed/index.html.erb:17]: Discussed with @naseberry and will leave in current form and raise a design ticket to review and update across the whole service

**Review by naseberry — APPROVED** [review]

**Review by colinbruce — APPROVED** [review]
Updates look fine :+1:

**Pull request merged**